api/resolvers/item.js

```diff
@@ -398,6 +400,27 @@ export default {
             orderBy: orderByClause(by || 'zaprank', me, models, type)
           }, ...whenRange(when, from, to || decodedCursor.time), decodedCursor.offset, limit, ...subArr)
           break
+        case 'random':
+          items = await itemQueryWithMeta({
+            me,
+            models,
+            query: `
+              ${selectClause(type)}
+              ${relationClause(type)}
+              ${whereClause(
+                '"Item"."deletedAt" IS NULL',
+                '"Item"."weightedVotes" > 10',
+                type === 'posts' && '"Item"."subName" IS NOT NULL',
+                subClause(sub, 5, subClauseTable(type), me, showNsfw),
+                typeClause(type),
+                await filterClause(me, models, type),
+                muteClause(me))}
+              ${orderByClause(by || 'random', me, models, type)}
+              OFFSET $3
+              LIMIT $4`,
+            orderBy: orderByClause(by || 'random', me, models, type)
+          }, ...whenRange(when, from, to || decodedCursor.time), decodedCursor.offset, limit, ...subArr)
+          break
```

Comment:
**Questions:**
Why is there a `whenClause` and then a `whenRange`?
Why are there two `orderBy`s?

I can't find what $1, $2 etc so left them as they are from `top` query

Reply:
So how come $3 and $4 can mean different things depending on the query?

Reply:
These are parameterized queries and the numbers correspond to parameter position: https://minervadb.xyz/how-parameterized-queries-and-bind-parameters-are-implemented-in-postgresql/

Reply:
Thanks for that. I actually am aware of names params but my question is more specific to the context of the app. I can't see how the params are being passed so that limit can correspond to $3 for one query and $4 for another one.

Reply:
It depends on where you put `limit` in the parameter list.

```js
}, ...whenRange(when, from, to || decodedCursor.time), decodedCursor.offset, limit, ...subArr)
```

Above, limit is the `$4`th parameter because `whenRange` produces two paramters.